src/vm/exceptionhandling.cpp

```diff
@@ -441,13 +441,24 @@ void ExceptionTracker::UpdateNonvolatileRegisters(CONTEXT *pContextRecord, REGDI
         }                                                                                   \
     } while (0)
 
+#define UPDATEREG_VAL(reg)                                                                  \
+    do {                                                                                    \
+        STRESS_LOG2(LF_GCROOTS, LL_INFO100, "Updating " #reg " %p to %p\n",                 \
+                pContextRecord->reg,                                                        \
+                pRegDisplay->pCurrentContext->reg);                                         \
+        pContextRecord->reg = pRegDisplay->pCurrentContext->reg;                            \
+    } while (0)
+
+
 #if defined(_TARGET_X86_)
 
     UPDATEREG(Ebx);
     UPDATEREG(Esi);
     UPDATEREG(Edi);
     UPDATEREG(Ebp);
 
+    UPDATEREG_VAL(ResumeEsp);
```

Comment:
I am not excited about the updating of ResumeEsp at this place since it is a not a non-volatile register and the function is called UpdateNonvolatileRegisters. Unfortunately the only other place where we have the necessary data to update the ResumeESP is the caller and that's called FixNonvolatileRegisters, so it would not be any better. And I don't really want to rename the functions.